infra/src/main/kotlin/fi/fta/geoviite/infra/linking/LinkingDao.kt

```diff
@@ -282,13 +283,14 @@ class LinkingDao(jdbcTemplateParam: NamedParameterJdbcTemplate?) : DaoBase(jdbcT
                  when location_track.topology_end_switch_id = :switch_id and segment.segment_index = alignment.segment_count-1 then 1
                  else 0
                end as end_is_joint,
-               postgis.st_x(postgis.st_startpoint(segment.geometry)) as start_x,
-               postgis.st_y(postgis.st_startpoint(segment.geometry)) as start_y,
-               postgis.st_x(postgis.st_endpoint(segment.geometry)) as end_x,
-               postgis.st_y(postgis.st_endpoint(segment.geometry)) as end_y
+               postgis.st_x(postgis.st_startpoint(segment_geometry.geometry)) as start_x,
```

Comment:
Siellä missä kysely oikeasti käyttää geometriaa, pitää nyt joinata tietenkin myös se. Aika rajallinen määrä näitä paikkoja